apps/plugin-browser/src/scripts/background/infer-entities.ts

```diff
@@ -0,0 +1,76 @@
+import { VersionedUrl } from "@blockprotocol/graph";
+import { ProposedEntity } from "@local/hash-isomorphic-utils/graphql/api-types.gen";
+
+import {
+  setErroredBadge,
+  setLoadingBadge,
+  setSuccessBadge,
+} from "../../shared/badge";
+import { InferEntitiesRequest } from "../../shared/messages";
+import { queryApi } from "../../shared/query-api";
+import { setInSessionStorage } from "../../shared/storage";
+
+const inferEntitiesQuery = /* GraphQL */ `
+  mutation inferEntities(
+    $textInput: String!
+    $entityTypeIds: [VersionedUrl!]!
+  ) {
+    inferEntities(
+      allowEmptyResults: false
+      entityTypeIds: $entityTypeIds
+      maxTokens: 0
+      model: "gpt-4-0613"
+      temperature: 0
+      textInput: $textInput
+      validation: PARTIAL
+    ) {
+      entities {
+        entityId
+        entityTypeId
+        properties
+        linkData {
+          leftEntityId
+          rightEntityId
+        }
+      }
+    }
+  }
+`;
+
+const inferEntitiesApiCall = (
+  textInput: string,
+  entityTypeIds: VersionedUrl[],
+) => {
+  return queryApi(inferEntitiesQuery, {
+    entityTypeIds,
+    textInput: textInput.slice(0, 7900),
+  }).then(
+    ({ data }: { data: { inferEntities: { entities: ProposedEntity[] } } }) => {
+      return data.inferEntities.entities;
+    },
+  );
+};
+
+export const inferEntities = async (message: InferEntitiesRequest) => {
+  void setInSessionStorage("inferenceStatus", { status: "pending" });
+  setLoadingBadge();
+
+  try {
+    const inferredEntities = await inferEntitiesApiCall(
+      message.textInput,
+      message.entityTypeIds,
+    );
```

Comment:
I'm not sure if there's a limitation I'm not thinking of, but would it be possible to slice the text here into chunks if it exceeds the maximum supported length (i.e. 7900 character), make multiple calls to `inferEntitiesApiCall` and then combine the results?

Reply:
We could, although we'd have to do some deduplication of the suggested entities (based on likely name properties) in case they appeared in multiple chunks. I'd rather leave this for follow up improvements once we've used it more.